base/stat.jl

```diff
@@ -439,29 +439,37 @@ operm(st::StatStruct) = UInt8((filemode(st)     ) & 0x7)
 
 # mode predicate methods for file names
 
+# The documentation in these files refer to taking the same argument as `stat`...
+for f in Symbol[
+    :uperm,
+    :gperm,
+    :operm,
+    :filemode,
+    :filesize,
+    :mtime,
+    :ctime,
+]
+    @eval ($f)(path...)  = ($f)(stat(path...))
+end
+
+# ...whereas these files should really only take paths (and not e.g. Ints)
+# See issue #51710
 for f in Symbol[
-    :ispath,
     :isfifo,
     :ischardev,
     :isdir,
     :isblockdev,
+    :ispath,
     :isfile,
     :issocket,
     :issetuid,
     :issetgid,
     :issticky,
-    :uperm,
-    :gperm,
-    :operm,
-    :filemode,
-    :filesize,
-    :mtime,
-    :ctime,
 ]
-    @eval ($f)(path...)  = ($f)(stat(path...))
+    @eval ($f)(path::AbstractString) = ($f)(stat(path))
```

Comment:
Should this keep the `...`?
```julia
@eval ($f)(path::AbstractString...) = ($f)(stat(path...))
```

Reply:
Probably also take at least one path; `isfile()` doesn't really make sense.

Reply:
Mmmm, `(path0::AbstractString, paths::AbstractString...)` or similar seems like a good idea.